Server/src/services/fileUploadService.ts

```diff
@@ -1,118 +1,129 @@
 const fileSystem = require('fs');
-const uploadModel = require('../models/DataUploadModel');
+import { DataUploadModel } from '../models/DataUploadModel'
 
 
 /**
  * The methods in this class are only responsible for processing uploaded files. Input will be parsed 
  * then stored into its appropriate table in the database. 
  */
 
-const processUpload = async (filePathOfJson) => {
+export class fileUploadService {
+  protected inputFile: string;
+  private uploadModel: DataUploadModel
 
-  let response = jsonUpload(filePathOfJson);
-  return response;
-}
+  constructor(inputFile: string) {
+    this.inputFile = inputFile;
+  }
 
-const jsonUpload = async (filePathOfJson) => {
+  async processUpload() {
 
-  let category = '';
-  let subcategory = '';
-  let dataSetName = '';
-  let dataType = '';
-  let dataSetComments = '';
-  let individualDataSetComments = [];
-  let material = [];
-  let referenceType = '';
-  let referencePublisher = '';
-  let referenceTitle = '';
-  let referenceAuthors = [];
-  let referenceYear;
-  let referencePages;
-  let referenceVolume;
+    let response = await this.jsonUpload(this.inputFile);
+    return response;
+  }
 
-  let jsonObj = (JSON.parse(fileSystem.readFileSync(filePathOfJson)));
+  private async jsonUpload(filePathOfJson: string) {
```

Comment:
I see that this function returns a string at the end, that should be included in function signature: 
private async jsonUpload(filePathOfJson: string): string {
...
}

Reply:
A+